WP_Auth0.php

```diff
@@ -352,19 +358,26 @@ public static function render_back_to_auth0() {
 
 	}
 
+	/**
+	 * Enqueue styles and scripts on the wp-login.php page if the plugin has been configured
+	 */
 	public function render_auth0_login_css() {
-		$client_id = WP_Auth0_Options::Instance()->get( 'client_id' );
-
-		if ( trim( $client_id ) === '' ) {
+		if ( ! WP_Auth0::ready() ) {
 			return;
 		}
-?>
-		<link rel='stylesheet' href='<?php echo plugins_url( 'assets/css/login.css', __FILE__ ); ?>' type='text/css' />
-	<?php
+
+		wp_enqueue_style( 'auth0', WPA0_PLUGIN_CSS_URL . 'login.css', FALSE, WPA0_VERSION );
+
+		if ( $this->a0_options->get( 'auth0_implicit_workflow' ) ) {
+			wp_enqueue_script( 'auth0-implicit', WPA0_PLUGIN_JS_URL . 'implicit-login.js', FALSE, WPA0_VERSION );
+			wp_localize_script( 'auth0-implicit', 'wpAuth0ImplicitGlobal', array(
+				'postUrl' => add_query_arg( 'auth0', 'implicit', site_url( 'index.php' ) ),
+			) );
+		}
 	}
 
 	public function render_form( $html ) {
-		if ( isset( $_GET['action'] ) && $_GET['action'] == 'lostpassword' ) {
+		if ( ( isset( $_GET['action'] ) && $_GET['action'] == 'lostpassword' ) || ! empty( $_GET[ 'auth0' ] ) ) {
```

Comment:
what does this change represent?

Reply:
If the `auth0` URL param is set, don't load the form. It's only sent when coming back from an implicit grant. Auth code grant goes straight to `index.php?auth0=1`

Reply:
If I load the form, I get a new `state` set so I don't want to do that ... also, no need to load the form at all since we're just passing through.

Reply:
Alright, I remember the `auth0` param now. Are you planning to refactor this "auth0 hack" anytime soon?

Reply:
The param can show up on the login page and on the `index.php` entry point.

Reply:
Not loading the form in this case does solve the problem of re-setting a state cookie but we don't need to load the login form at all if this is a response from Auth0. So, I would say that the URL parameter check will stay but state checking will evolve to handle this better (by abstracting that `WP_Auth0_Nonce_Handler` class to handle both scenarios)